src/RoadRegistry.Syndication.ProjectionHost/AtomFeedProcessor.cs

```diff
@@ -0,0 +1,259 @@
+                                    var context = dbContextFactory();
+                                    context.ChangeTracker.AutoDetectChangesEnabled = false;
+
+                                    var entries = (await reader.ReadEntriesAsync(
+                                            feedConfiguration.Uri,
+                                            catchUpPosition,
+                                            feedConfiguration.UserName,
+                                            feedConfiguration.Password,
+                                            true,
+                                            false))
+                                        .ToList();
+
+                                    while (entries.Any())
+                                    {
+                                        if (!long.TryParse(entries.Last().Id, out var lastEntryId))
+                                            break;
+
+                                        foreach (var atomEntry in entries)
+                                        {
+                                            // if (catchUp.AfterPosition.HasValue &&
+                                            //     streamMessage.Position == catchUp.AfterPosition.Value)
+                                            // {
+                                            //     continue; // skip already processed message
+                                            // }
+
+                                            // if (filter(streamMessage))
+                                            if (true) // filter here?
+                                            {
+
+                                                logger.LogInformation("[{Context}] Catching up on {MessageType} at {Position}",
+                                                            typeof(TSyndicationContent).Name, atomEntry.ContentType, atomEntry.Id);
+
+                                                var envelope = envelopeFactory.CreateEnvelope<TSyndicationContent>(atomEntry);
+                                                if (envelope != null)
+                                                {
+                                                    var handlers = resolver(envelope);
+                                                    foreach (var handler in handlers)
+                                                    {
+                                                        await handler
+                                                            .Handler(context, envelope, _messagePumpCancellation.Token)
+                                                            .ConfigureAwait(false);
+                                                    }
+                                                }
+                                            }
+
+                                            observedMessageCount++;
+                                            catchUpPosition = lastEntryId;
```

Comment:
I'd be inclined to keep updating the position inside the loop - saves you from querying in advance.